### PR TITLE
Sprint 4: 4-LoginSignupValidation-GreydonOKeefe

### PR DESCRIPTION
**Pull request by mcho9434**
New Changes:
- Toasts for a variety of actions
- new userByEmail endpoint

React-Toastify is a new dependency, just makes toast easy

concerns: Greydon's local dev has some issues, would you mind j running this on another machine? Works on mine.

### CODE REVIEW

**Comment by mcho9434** [comment]
@oliviasl 
Submitting for Greydon

File: client/src/routes/HomePage.tsx [hunk 1]
```diff
@@ -1,68 +1,270 @@
...
+
+  const authUser = async () => {
+    // invalid email toast condition
+    if (!email.includes("@")) {
+      toast.warning("Email format is invalid.");
+      return;
+    }
+
+    const authBody =  JSON.stringify({
+      email: email,
```
Comment (oliviasl) [inline client/src/routes/HomePage.tsx:95]: I'm getting a JSON formatting error here so there might be something weird going on with the info getting passed in

File: client/src/routes/HomePage.tsx [hunk 2]
```diff
@@ -1,68 +1,270 @@
...
+    const response = await fetch("/api/validateUser", {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: authBody
+    });
+    const status = await response.json();
+    if (status <= 0) {
```
Comment (oliviasl) [inline client/src/routes/HomePage.tsx]: The status is returned in the format [user_id, user_status] so you may need to use something like status[1] <= 0 instead

**Review by oliviasl — CHANGES_REQUESTED** [review]
Good work so far! The toasts function perfectly but they don't match the styles right now, could you try to override some of the react toast styles? Also I left some comments on the code because the toast for unauthorized users doesn't work for me right now, so please take a look at that too.

**Comment by mcho9434** [comment]
@greydonokeefe Wanna take a look and lmk?

**Comment by mcho9434** [comment]
@oliviasl

**Pull request closed without merging**